### ISSUE TITLE
Unroll shows wrong row index

### ISSUE THREAD

**Issue by JohnnyGrande** [renatoathaydes/spock-reports#70]
When using @Unroll with the following test, we'd expect indices 2 and 4 to be the failures...  what we get is much different.

### Test:
```gradle
import spock.lang.Specification
import spock.lang.Unroll

class MaxSpec extends Specification {
    @Unroll
    def 'Get the max value of two numbers'() {
        expect: 'Should return the bigger number'
        Math.max(a, b) == c

        where: 'max(a,b) == c'
        a | b || c
        1 | 0 || 1
        2 | 3 || 3
        3 | 1 || 1
        4 | 2 || 4
        2 | 3 || 2
    }
}
```

### Result:
![screen shot 2016-05-26 at 4 17 10 pm](https://cloud.githubusercontent.com/assets/14222093/15593032/5eb1337c-235d-11e6-9c8e-fe298265a9f9.png)

**Comment by renatoathaydes** [comment]
Thanks for reporting the bug. The order of the failures are reported correctly, it seems only the indexes got messed up... will fix asap.

**Comment by renatoathaydes** [comment]
The names of the features are provided by Spock, spock-reports just reads it. Looks like every time it reads it, the index is increased :( I think I will have to read the names once and cache them... but it seems that even the first read already gets the wrong index, as it starts at `5` (presumably indexes 0 to 4 were read by spock itself).

**Comment by renatoathaydes** [comment]
OK, confirmed the behaviour is as described above and it's happening at this class (from Spock): `org.spockframework.runtime.extension.builtin.UnrollNameProvider`. I will need to use this class differently to get the name of the feature, or use something else for that.

**Comment by renatoathaydes** [comment]
@JohnnyGrande do you have time to pull the branch called `issue68` and confirm that this is fixed after my commit above?

**Comment by tilmanginzel** [comment]
Works for me. Same spec as above:
![spock-unroll-result](https://cloud.githubusercontent.com/assets/3899565/15922968/cc77639a-2e2a-11e6-87e5-f94a813b1ff6.png)

**Comment by renatoathaydes** [comment]
@tilmanginzel excellent to know. Thanks.

**Comment by renatoathaydes** [comment]
Both this issue and issue #68 have been fixed in release 1.2.12, which I have just pushed.

**Comment by tilmanginzel** [comment]
Perfect, thank you! :+1:
